File: app/src/main/java/de/stephanlindauer/criticalmaps/model/StorageLocation.java
package de.stephanlindauer.criticalmaps.model;

import android.os.StatFs;
import java.io.File;


public class StorageLocation {
    private final File dbFile;

    public StorageLocation(File path) {
        if (!path.exists()) {
            throw new IllegalArgumentException("Path does not exist or is read only: " + path);
        }

        dbFile = new File(path + File.separator + "mbgl-offline.db");
    }

    public long getFreeSpaceBytes() {
        return new StatFs(dbFile.getAbsolutePath()).getAvailableBytes();
    }

    public long getTotalSizeBytes() {
        return new StatFs(dbFile.getAbsolutePath()).getTotalBytes();
    }

    public long getUsedSpace() {
        return getTotalSizeBytes() - getFreeSpaceBytes();
    }

    public long getCacheSize() {
        long cacheSize = 0;
        if (dbFile.exists()) {
            cacheSize = dbFile.length();
        }

        return cacheSize;
    }
}

File: app/src/main/java/de/stephanlindauer/criticalmaps/model/StorageLocation.kt
package de.stephanlindauer.criticalmaps.model

import android.os.StatFs
import java.io.File

class StorageLocation(path: File) {
    private val dbFile: File

    init {
        require(path.exists()) { "Path does not exist or is read only: $path" }
        dbFile = File(path, "mbgl-offline.db")
    }

    val freeSpaceBytes: Long
        get() = StatFs(dbFile.absolutePath).availableBytes

    val totalSizeBytes: Long
        get() = StatFs(dbFile.absolutePath).totalBytes

    val usedSpace: Long
        get() = totalSizeBytes - freeSpaceBytes

    val cacheSize: Long
        get() = if (dbFile.exists()) dbFile.length() else 0
}
